migrations/back_populate_source_plate_and_sample_uuids.py

```diff
@@ -222,143 +233,145 @@ def update_uuids_mongo_and_mlwh(config: Config, source_plate_barcodes: List[str]
-            },
-            update={
-                "$set": {
-                    FIELD_LH_SAMPLE_UUID: sample_doc[FIELD_LH_SAMPLE_UUID],
-                    FIELD_LH_SOURCE_PLATE_UUID: sample_doc[FIELD_LH_SOURCE_PLATE_UUID],
-                    FIELD_UPDATED_AT: sample_doc[FIELD_UPDATED_AT],
-                }
-            },
-        )
-        if mongo_sample is None:
-            return False
-        else:
-            return True
+    log_mongo_sample_fields("Before update", [r[RECORD_KEY_ORIGINAL] for r in records])
 
-    except Exception as e:
-        LOGGER.critical("Failed to update sample in mongo for mongo id " f"{sample_doc[FIELD_MONGODB_ID]}")
-        LOGGER.exception(e)
-    return False
+    with mongo_db.client.start_session() as session:
+        with session.start_transaction():
+            samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
 
+            for record in records:
+                update_doc = record[RECORD_KEY_UPDATED]
 
-def update_mlwh_sample_uuid_and_source_plate_uuid(mysql_conn: MySQLConnectionAbstract, sample_doc: SampleDoc) -> bool:
-    """Updates a sample in the sample in the MLWH database
+                try:
+                    mongo_sample = samples_collection.find_one_and_update(
+                        filter={
+                            FIELD_MONGODB_ID: update_doc[FIELD_MONGODB_ID],
+                        },
+                        update={
+                            "$set": {
+                                FIELD_LH_SAMPLE_UUID: update_doc[FIELD_LH_SAMPLE_UUID],
+                                FIELD_LH_SOURCE_PLATE_UUID: update_doc[FIELD_LH_SOURCE_PLATE_UUID],
+                                FIELD_UPDATED_AT: update_doc[FIELD_UPDATED_AT],
+                            }
+                        },
+                    )
 
-    Arguments:
-        mysql_conn {MySQLConnectionAbstract} -- a connection to the MLWH MySQL database
-        sample_doc {SampleDoc} -- the sample document whose fields should be updated
+                    if mongo_sample is None:
+                        raise MongoUpdateError("No document was returned for find_one_and_update.")
 
-    Returns:
-        bool -- whether the updates completed successfully
+                except Exception as e:
+                    LOGGER.critical("Failed to update sample in mongo for mongo id " f"{update_doc[FIELD_MONGODB_ID]}")
+                    LOGGER.exception(e)
+                    session.abort_transaction()
+                    raise
+
+            session.commit_transaction()
+
+    log_mongo_sample_fields("After successful update", [r[RECORD_KEY_UPDATED] for r in records])
+
+
+def log_mlwh_sample_fields(description, mlwh_samples):
+    DATA_LOGGER.info(f"=== Logging MLWH sample fields -- {description} ===")
+    for sample in mlwh_samples:
+        DATA_LOGGER.info(sample)
+    DATA_LOGGER.info(f"=== End of MLWH sample fields -- {description} ===")
```

Comment:
Seems duplicate of log_mongo_sample_fields... maybe can be refactored

Reply:
Yeah I noticed that as well and thought about combining them.  Probably should have done.